Partially revert https://gitlab.com/veloren/veloren/-/commit/a698a7d46ed7
until lang/rust >= 1.80 update

error[E0658]: use of unstable library feature 'option_take_if'
    --> common/src/comp/inventory/item/mod.rs:1622:22
     |
1622 |                     .take_if(|contained_item| matches!(contained_item.quality(), Quality::Debug));
     |                      ^^^^^^^
     |
     = note: see issue #98934 <https://github.com/rust-lang/rust/issues/98934> for more information
     = help: add `#![feature(option_take_if)]` to the crate attributes to enable
     = note: this compiler was built on 2024-06-10; consider upgrading it if it is out of date

--- common/src/lib.rs.orig	2024-07-10 22:07:53 UTC
+++ common/src/lib.rs
@@ -7,6 +7,7 @@
     associated_type_defaults,
     fundamental,
     let_chains,
+    option_take_if,
     option_zip,
     trait_alias,
     type_alias_impl_trait,
